addon/components/document-delete-button.js

```diff
@@ -30,24 +30,31 @@ export default class DocumentDeleteButtonComponent extends Component {
     }
   }
 
-  @task *delete() {
+  delete = task(async () => {
     try {
       if (this.args.docsToDelete) {
         const docs = Array.isArray(this.args.docsToDelete)
           ? this.args.docsToDelete
           : [this.args.docsToDelete]; // if the supplied argument is not an array we make it one
 
-        for (const doc of docs) {
-          yield doc.destroyRecord().catch((error) => {
-            doc.rollbackAttributes();
-            throw error;
-          });
-          this.documents.deselectDocument(doc);
+        if (this.args.onConfirm) {
+          this.args.onConfirm(docs);
         }
-      }
 
-      if (this.args.onConfirm) {
-        this.args.onConfirm(this.args.docsToDelete);
+        const deletionStatus = await Promise.allSettled(
+          docs.map((doc) => {
+            return doc.destroyRecord();
+          }),
+        );
+
+        deletionStatus.forEach(({ status, reason }, index) => {
+          if (status === "rejected") {
```

Comment:
filter instead for rejected responses, rollback them all, and finally throw.

Reply:
Done